tutor/src/flux/course-enrollment.coffee

```diff
@@ -0,0 +1,102 @@
+# coffeelint: disable=max_line_length
+
+_ = require 'underscore'
+React = require 'react'
+EventEmitter2 = require 'eventemitter2'
+
+ERROR_MAP = require './course-enrollment-handled-errors'
+
+CourseEnrollmentActions = flux.createActions [
+  'create'
+  'created'
+  'approve'
+  'approved'
+  'FAILED'
+]
+
+CourseEnrollmentStore = flux.createStore
+  actions: _.values(CourseEnrollmentActions)
+  status: "loading"
+
+  exports:
+    # Still loading (creating the EnrollmentChange)
+    isLoading: -> @status is "loading"
+
+    # A registration has been created, but not confirmed
+    isPending: -> @status is "pending"
+
+    # complete and ready for use
+    isRegistered: -> @status is "approved"
+
+    getStudentIdentifier: -> ""
+
+    courseId: -> @to?.course?.id
+
+    confirm: (studentId) -> CourseEnrollmentActions.approve(@id, studentId)
+
```

Comment:
I see... my issue here is that the shared code does not use flux (because the CC code doesn't) and so it doesn't distinguish between stores and actions...